File: mau/a1271.cpp
#include <iostream>
using namespace std;
int a[1000], n;
int main()
{
    int i, dmax=0, d=1,v;
    cin>>n;
    for (i=1; i<=n; i++) cin>>a[i];
    for (i=2; i<=n; i++) {
        if (a[i] > a[i-1]) d++;
        else {
            if (d > dmax) {
                dmax = d;
                v = i - d;
            }
            d = 1;
        }
    }
    if (d > dmax) {
        dmax = d;
        v = n+1-d;
    }
    for (i=v; i<v+dmax; i++) cout<<a[i]<<" ";
    return 0;
}

File: mau/a1271.pas
var n, i, dmax, d, v : integer;
    a : array[1..1000] of integer;
BEGIN
    readln(n);
    for i:= 1 to n do read(a[i]);
    dmax:=0;
    d:=1;
    for i:= 2 to n do begin
        if a[i] > a[i-1] then inc(d)
        else begin
            if d > dmax then begin
                dmax:=d;
                v:= i-d;
            end;
            d:=1;
        end;
    end;
    if d > dmax then begin
        dmax:=d;
        v:=n+1-d;
    end;
    for i:= v to v+dmax-1 do write(a[i],' ');
END.
